js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useCallback, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import './useCroppedImageSelector.scss';
+
+/**
+ * @typedef {import('.~/hooks/types.js').ImageMedia} ImageMedia
+ */
+
+/**
+ * @typedef {Object} CroppedImageSelector
+ * @property {(preselectedId: number) => void} openSelector Function to open the selector modal.
+ */
+
+/**
+ * Calculates the maximum cropping size according to the fixed aspect ratio.
+ *
+ * @param {number} width The image width.
+ * @param {number} height The image height.
+ * @param {number} widthScale The scale of width.
+ * @param {number} heightScale The scale of height.
+ * @return {[number, number]} The tuple of cropped width and height.
+ */
+export function cropByFixedRatio( width, height, widthScale, heightScale ) {
+	const ratio = widthScale / heightScale;
+
+	if ( width / height > ratio ) {
+		width = Math.round( height * ratio );
+	} else {
+		height = Math.round( width / ratio );
+	}
+
+	return [ width, height ];
+}
+
```

Comment:
💅  Even tho you added the refs of the extended library. I missed some explanations on each file function, just explaining what we are doing there is enough, no need to be super detailed. I guess tat would save time of checking parent files for the developers landing on this code.

Reply:
Since the WP Media library seems to have no documents and it's mainly built on top of Backbone.js and mixed some WP custom mechanisms,  the need for additional explanations very depends on the readers' understanding of this background knowledge. Could you help point out which places need more explanation?

Reply:
>  Could you help point out which places need more explanation?

Especially I had doubts about these 2. (the second one in L171, for sure)


- In the chunk of code for handling the invalid size and showing error message: (Line 216)

maybe something like; 
// Check if the selected image has a valid width and height. If it has invalid date show an error message in the toolbar. 


- Line 171, about the `states`. Maybe explain why we are loading here this controllers. 

// Maybe like override the WP Media adding suggested width and height for the image and a custom image cropper. 

(Unsure if its accurate)  

Optionally to document the handlers and other internal functions.

Reply:
Not sure line numbers 171 and 216 are referred to which code revision. Based on the comment content, I guess it means:
- https://github.com/woocommerce/google-listings-and-ads/blob/54da6a7d95648b4b0f8f7f8576960699fc1420df/js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js#L233
- https://github.com/woocommerce/google-listings-and-ads/blob/54da6a7d95648b4b0f8f7f8576960699fc1420df/js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js#L198

I added some code comments and adjusted some naming in https://github.com/woocommerce/google-listings-and-ads/pull/1819/commits/bd25f054a082d990ea94d76759ceac193bd52ef9. Most internal functions still don't have additional documents because the function names should be sufficient to describe themselves. Hope it has covered where the explanations were needed.